File: Python/python-mini-games/Tic_tac_toe/tic_tac-toe.py
'''Player:1 Is "X"
   Player:2 Is "O"'''


board = [
         ['0','1','2'],
         ['3','4','5'],
         ['6','7','8']
        ]

def print_board():
  for i in board:
    print(i)


def p1():

  n = int(input("Player 1_move.  "))
  row = n//3
  column = n%3
  board[row][column]="X"
  print_board()

def p2():

  n = int(input("Player 2_move.  "))
  row = n//3
  column = n%3
  board[row][column]="O"
  print_board()

def end():
  if board[0][0]==board[1][0]==board[2][0]:
    return(board[0][0])

  elif board[0][1]==board[1][1]==board[2][1]:
    return(board[0][1])

  elif board[0][2]==board[1][2]==board[2][2]:
    return(board[0][2])


  elif board[0][0]==board[0][1]==board[0][2]:
    return(board[0][0])

  elif board[1][0]==board[1][1]==board[1][2]:
    return(board[1][1])

  elif board[2][0]==board[2][1]==board[2][2]:
    return(board[2][0])


  elif board[0][0]==board[1][1]==board[2][2]:
    return(board[0][0])

  elif board[0][2]==board[1][1]==board[2][0]:
    return(board[0][2])

  else:
    return("draw")




print_board()
count=0
for i in range(5):
  p1()
  count+=1
  a = end()
  if a=="X":
    print("Game Ended")
    print("Player 1")
    break
  if count==9:
    print("draw")
    break

  p2()
  count+=1
  a = end()
  if a=="O":
    print("Game Ended")
    print("Player 2")
    break

File: cpp/tic-tac-toe.cpp
#include<stdio.h>
#include<stdlib.h>

void initializeBoard();
int DeterminingOutcome();
char Board[10]={'0','1','2','3','4','5','6','7','8','9'};

int main()
{
	int currentPlayer=1;
	int playerIndex=-1;
	int cellNumberToChange;
	
	char playInput;
	
	do
	{
		initializeBoard();
		
		currentPlayer = (currentPlayer%2)? 1:2;
		
		printf("player %d, enter a number:", currentPlayer);
		
		scanf("%d", &cellNumberToChange);
		
		playInput = (currentPlayer == 1 ) ? 'X':'O';
		
		if(cellNumberToChange == 1 && Board[1] == '1')
		{
			Board[1] = playInput;
		}
		
		else if(cellNumberToChange == 2 && Board[2] == '2')
		{
			Board[2] = playInput;
		}
		else if(cellNumberToChange == 3 && Board[3] == '3')
		{
			Board[3] = playInput;
		}
		else if(cellNumberToChange == 4 && Board[4] == '4')
		{
			Board[4] = playInput;
		}
		else if(cellNumberToChange == 5 && Board[5] == '5')
		{
			Board[5] = playInput;
		}
		else if(cellNumberToChange == 6 && Board[6] == '6')
		{
			Board[6] = playInput;
		}
		else if(cellNumberToChange == 7 && Board[7] == '7')
		{
			Board[7] = playInput;
		}
		else if(cellNumberToChange == 8 && Board[8] == '8')
		{
			Board[8] = playInput;
		}
		else if(cellNumberToChange == 9 && Board[9] == '9')
		{
			Board[9] = playInput;
		}
		else
		{
			printf("INVALID MOVE!!!");
			currentPlayer--;
			
			getch();
		}
		
		playerIndex= DeterminingOutcome();
		currentPlayer++;
		
	} while (playerIndex == -1);
	
	InitializeBoard();
	
	if (playerIndex == 1)
	{
		printf("\aPlayer %d won the game!", --currentPlayer);
	}
	else
	{
		printf("game is a draw!");
	}
	getch();
	
}

void initializeBoard()
{
	
	system("cls");
	printf("\n\n\tTic-Tac-Toe 2-Player \n\n");
	
	printf("Player 1 plays as X\n");
	printf("Player 2 plays as O\n\n\n");
	
	printf("	|	|	|\n");
	
	printf(" %c | %c | %c |\n", Board[1], Board[2], Board[3]);
	
	printf("____|____|____|\n");
	printf("	|	 |	  |\n");
	
	printf(" %c | %c | %c |\n", Board[4], Board[5], Board[6]);
	
	printf("____|____|____|\n");
	printf("	|	 |	  |\n");
	
	printf(" %c | %c | %c |\n", Board[7], Board[8], Board[9]);
	
	printf("	|	|	|\n");
	
}

int DeterminingOutcome()
{
	if(Board[1] == Board[2] && Board[2]==Board[3])
	{
		return 1;
	}
	
	else if(Board[4] == Board[5] && Board[5]==Board[6])
	{
		return 1;
	}
	else if(Board[7] == Board[8] && Board[8]==Board[9])
	{
		return 1;
	}
	else if(Board[1] == Board[4] && Board[4]==Board[7])
	{
		return 1;
	}
	else if(Board[2] == Board[5] && Board[5]==Board[8])
	{
		return 1;
	}
	else if(Board[3] == Board[6] && Board[6]==Board[9])
	{
		return 1;
	}
	else if(Board[1] == Board[5] && Board[5]==Board[9])
	{
		return 1;
	}
	else if(Board[3] == Board[5] && Board[5]==Board[7])
	{
		return 1;
	}
	
	else if(Board[1] != '1' && Board[2] != '2' && Board[3] != '3' 
	&& Board[4] != '4' && Board[5] != '5' && Board[6] != '6'
	 && Board[7] != '7' && Board[8] != '8' && Board[9] != '9')
	
	{
		return 0;
	}
	
	else
	{
		return -1;
	}
	
}

